### PR TITLE
[16.0][mig] analytic_partner

### PR DESCRIPTION
**Pull request by dreispt**
Supersedes https://github.com/OCA/account-analytic/pull/581

### CODE REVIEW

File: analytic_partner/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#. Go to **Invoicing > Configuration > Management > Analytic Items**.
```
Comment (chienandalu) [inline analytic_partner/readme/USAGE.rst:1]:
```suggestion
#. Go to **Invoicing > Accounting > Management > Analytic Items**.
```

File: analytic_partner/models/account_move_line.py [hunk 2]
```diff
@@ -8,7 +8,8 @@
 class AccountMoveLine(models.Model):
     _inherit = "account.move.line"
 
-    def _prepare_analytic_line(self):
-        res = super()._prepare_analytic_line()
-        res[0]["other_partner_id"] = self.move_id.partner_id.commercial_partner_id.id
+    def _prepare_analytic_lines(self):
```
Comment (chienandalu) [inline analytic_partner/models/account_move_line.py:11]: I think that `_prepare_analytic_distribution_line` is the proper method to hook into now.

**Review by chienandalu — COMMENTED** [review]
HI @dreispt I'm struggling to find out why this module is still needed. Searching for previous internal issues the problem seemed to be that when the analytic account used had a partner set it took that one instead of the one in the move line...

But I can't reproduce that issue anymore. What case are we covering here right now? As far as I could test, the other partner is always the same as the partner.

File: analytic_partner/models/account_move_line.py (hunk 2 above)
Comment (maisim) [reply analytic_partner/models/account_move_line.py:11]: Hi @chienandalu , I think you are right, we can add `other_partner_id` to the dict returned by _prepare_analytic_distribution_line.
Somethings like:
```
def _prepare_analytic_distribution_line(self, distribution, account_id, distribution_on_each_plan):
    vals = super()._prepare_analytic_distribution_line(distribution, account_id, distribution_on_each_plan)
    vals['other_partner_id'] = self.move_id.partner_id.commercial_partner_id.id
    return vals
```
Hi @dreispt, How do you want to proceed? Can I create a PR on your fork?

**Review by maisim — CHANGES_REQUESTED** [review]

**Comment by flotho** [comment]
Hi @maisim , thanks for review with @chienandalu 
Ad @dreispt doesn't seems to have new commits since a long time, I think you could use his branch to create a new PR  to supersedes this one. Authoring will be kept anyway.
Regards

**Comment by dreispt** [comment]
@maisim Yes, you are welcome to pull this branch into your fork and make your own PR.

**Comment by maisim** [comment]
There is another PR (a third) to migrate this module to 16.0 : https://github.com/OCA/account-analytic/pull/672
It seems almost finish \o/

Can someone update the Migration to 16.0 issue https://github.com/OCA/account-analytic/issues/489 and close  the current PR https://github.com/OCA/account-analytic/pull/609  ?

**Pull request closed without merging**